src/FTPC.jl

```diff
@@ -268,65 +256,59 @@ Download file with persistent connection.
-            end
+    if ~isempty(save_path)
+        wd.buffer = open(save_path, "w")
+    end
 
-            p_wd = pointer_from_objref(wd)
-            p_resp = pointer_from_objref(resp)
+    try
 
-            if ctxt.options.binary_mode
-                # We need to switch the url to point to the file directly
-                # There's no need for a command
-                p = ctxt.url * file_name
-                @ce_curl curl_easy_setopt CURLOPT_URL p
-            else
-                # RETR is stuck in ascii mode https://curl.haxx.se/mail/lib-2004-12/0219.html
-                command = "RETR " * file_name
-                @ce_curl curl_easy_setopt CURLOPT_CUSTOMREQUEST command
-            end
+        p_wd = pointer_from_objref(wd)
+        p_resp = pointer_from_objref(resp)
 
-            @ce_curl curl_easy_setopt CURLOPT_WRITEFUNCTION c_write_file_cb
-            @ce_curl curl_easy_setopt CURLOPT_WRITEDATA p_wd
+        if ctxt.options.binary_mode
+            # We need to switch the url to point to the file directly
+            # There's no need for a command
+            p = ctxt.url * file_name
+            @ce_curl curl_easy_setopt CURLOPT_URL p
+        else
+            # RETR is stuck in ascii mode https://curl.haxx.se/mail/lib-2004-12/0219.html
+            command = "RETR " * file_name
+            @ce_curl curl_easy_setopt CURLOPT_CUSTOMREQUEST command
+        end
 
-            @ce_curl curl_easy_setopt CURLOPT_HEADERFUNCTION c_header_command_cb
-            @ce_curl curl_easy_setopt CURLOPT_HEADERDATA p_resp
+        @ce_curl curl_easy_setopt CURLOPT_WRITEFUNCTION c_write_file_cb
+        @ce_curl curl_easy_setopt CURLOPT_WRITEDATA p_wd
 
-            @ce_curl curl_easy_perform
-            process_response(ctxt, resp)
+        @ce_curl curl_easy_setopt CURLOPT_HEADERFUNCTION c_header_command_cb
+        @ce_curl curl_easy_setopt CURLOPT_HEADERDATA p_resp
 
-            if ~isempty(save_path)
-                close(wd.buffer)
-            end
+        @ce_curl curl_easy_perform
+        process_response(ctxt, resp)
 
-            if isopen(wd.buffer)
-                seekstart(wd.buffer)
-            end
+        if isopen(wd.buffer)
+            seekstart(wd.buffer)
+        end
 
-            if ctxt.options.binary_mode
-                # We need to switch it back to the original url
-                @ce_curl curl_easy_setopt CURLOPT_URL ctxt.url
-            end
+        if ctxt.options.binary_mode
+            # We need to switch it back to the original url
+            @ce_curl curl_easy_setopt CURLOPT_URL ctxt.url
+        end
 
-            resp.bytes_recd = wd.bytes_recd
-            resp.body = wd.buffer
+        resp.bytes_recd = wd.bytes_recd
+        resp.body = wd.buffer
 
-            return resp
+        return resp
 
-        catch
-            cleanup_easy_context(ctxt)
-            rethrow()
+    catch
+        cleanup_easy_context(ctxt)
+        rethrow()
```

Comment:
Should cleanup happen here? Isn't this similar to closing a file you never opened?